src/editor/mod.rs

```diff
@@ -79,11 +81,21 @@ impl Editor {
                 RedrawEvent::SetTitle { title } => {
                     EVENT_AGGREGATOR.send(WindowCommand::TitleChanged(title));
                 }
-                RedrawEvent::ModeInfoSet { cursor_modes } => self.mode_list = cursor_modes,
+                RedrawEvent::ModeInfoSet { cursor_modes } => {
+                    self.mode_list = cursor_modes;
+                    if let Some(curr_mode_i) = self.current_mode_index.clone() {
```

Comment:
nit: please write out current, helps with somebody reading the code for the first time